### PR TITLE
Option docs improvements

### PR DESCRIPTION
**Pull request by roberth**
###### Description of changes

Some improvements I've found while working on the NixOS test framework (#176557). They're small fixes and quality of life improvements.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/lib/make-options-doc/mergeJSON.py [hunk 1]
```diff
@@ -57,19 +57,24 @@ def _get_method(self, name):
...
             return text + "\n\n"
+        def newline(self):
+            return "<literallayout>\n</literallayout>"
         def codespan(self, text):
-            return f"<literal>{text}</literal>"
+            return f"<literal>{escape(text)}</literal>"
         def block_code(self, text, info=None):
-            info = f" language={quoteattr(info)}" if info is not None else ""
-            return f"<programlisting{info}>\n{text}</programlisting>"
+            if info == "{=docbook}":
```
Comment (pennae) [inline nixos/lib/make-options-doc/mergeJSON.py:73]: not sure having that hatch is such a great thing. we'll have to make some adjustment to the documentation to get it to work as markdown anyway, and if we have a hatch we have one more place to keep in mind when we finally axe docbook.

do you have a special use in mind where this is necessary? maybe a few more extension to the renderer would be better.

**Review by pennae — COMMENTED** [review]
don't know enough about the xslt sheets to comment on those, but the rest looks good. :+1:

File: nixos/lib/make-options-doc/mergeJSON.py (hunk 1 above)
Comment (roberth) [reply nixos/lib/make-options-doc/mergeJSON.py:73]: I used it while I couldn't get `blockquote` accepted by the docbook tooling that runs after this. I suppose I could have used a plain old docbook description instead. I'll remove it.

**Comment by roberth** [comment]
> xslt sheets

No worries, I'm somewhat familiar with them, and I've tested the change.

**Pull request merged**